python/beeai_framework/tools/errors.py

```diff
@@ -29,6 +32,19 @@ def __init__(
     ) -> None:
         super().__init__(message, cause=cause, context=context)
 
+    @classmethod
+    def ensure(
+        cls,
+        error: Exception,
+        *,
+        message: str | None = None,
+        context: dict[str, Any] | None = None,
+        tool: Optional["AnyTool"] = None,
+    ) -> "FrameworkError":
+        tool_context = {"name": tool.name} if tool is not None else {}
+        context.update(tool_context) if context is not None else None
```

Comment:
This should be inverted. Instead of modifying the dict the user just passed, we should create a copy and update that copy.

Reply:
done